### PR TITLE
Merge 1.7.7.x into develop - 17/03/2020

### PR DESCRIPTION
**Pull request by matks**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | 
| Type?         | improvement
| Category?     | ME
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | 
| How to test?  | No need for QA

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/18177)

### CODE REVIEW

File: admin-dev/themes/new-theme/js/components/tinymce-editor.js [hunk 1]
```diff
@@ -163,26 +162,14 @@ class TinyMCEEditor {
...
 
         $(textareaLinkSelector, tabContainer).on('shown.bs.tab', () => {
-          const form = $(textarea).closest('form');
           const editor = window.tinyMCE.get(textarea.id);
           if (editor) {
             // Reset content to force refresh of editor
             editor.setContent(editor.getContent());
           }
-
-          EventEmitter.emit('languageSelected', {
```
Comment (matks) [inline admin-dev/themes/new-theme/js/components/tinymce-editor.js:173]: @NeOMakinG I re-applied this https://github.com/PrestaShop/PrestaShop/commit/99cfdbf1fb8c7d50db1cf1ec0df9c2638ac2c018

Can you confirm I did it well ?

File: admin-dev/themes/new-theme/js/pages/order/create/create-order-page.js [hunk 2]
```diff
@@ -66,6 +66,46 @@ export default class CreateOrderPage {
...
+   *
+   * @returns {boolean}
+   */
+  static validateSelectedAddresses(addresses) {
+    let deliveryValid = false;
+    let invoiceValid = false;
+
+    const keys = Object.keys(addresses);
+
+    for (let i = 0; i < keys.length; i += 1) {
```
Comment (matks) [inline admin-dev/themes/new-theme/js/pages/order/create/create-order-page.js:92]: Add to adapt this for ESLint to be happy
@NeOMakinG Can you confirm I did well translating `for (const key in addresses) {` into this ?

Previous statement: https://github.com/PrestaShop/PrestaShop/blob/1.7.7.x/admin-dev/themes/new-theme/js/pages/order/create/create-order-page.js#L89

File: admin-dev/themes/new-theme/js/pages/order/create/create-order-page.js [hunk 3]
```diff
@@ -488,4 +562,32 @@ export default class CreateOrderPage {
...
+  refreshAddressesList(refreshCartAddresses) {
+    const cartId = $(createOrderMap.cartBlock).data('cartId');
+    $.get(this.router.generate('admin_carts_info', {cartId})).then((cartInfo) => {
+      this.addressesRenderer.render(cartInfo.addresses);
+
+      if (refreshCartAddresses) {
+        this.changeCartAddresses();
+      }
+    }).catch((e) => {
+      window.showErrorMessage(e.responseJSON.message);
```
Comment (matks) [inline admin-dev/themes/new-theme/js/pages/order/create/create-order-page.js:582]: @NeOMakinG `showErrorMessage()` => `window. showErrorMessage()` is that right ?

**Comment by matks** [comment]
Need @NeOMakinG confirmation as ESLint required me to modify multiple JS files

File: admin-dev/themes/new-theme/js/components/tinymce-editor.js (hunk 1 above)
Comment (NeOMakinG) [reply admin-dev/themes/new-theme/js/components/tinymce-editor.js:173]: Yes it looks fine

File: admin-dev/themes/new-theme/js/pages/order/create/create-order-page.js (hunk 2 above)
Comment (NeOMakinG) [reply admin-dev/themes/new-theme/js/pages/order/create/create-order-page.js:92]: Looks fine, even if initially I would use a .forEach, that's fine like this

File: admin-dev/themes/new-theme/js/pages/order/create/create-order-page.js (hunk 3 above)
Comment (NeOMakinG) [reply admin-dev/themes/new-theme/js/pages/order/create/create-order-page.js:582]: What do you mean ? Didn't understand this one

File: install-dev/upgrade/sql/1.7.8.0.sql [hunk 4]
```diff
@@ -2,7 +2,6 @@ SET SESSION sql_mode='';
 SET NAMES 'utf8mb4';
 
 INSERT IGNORE INTO `PREFIX_hook` (`id_hook`, `name`, `title`, `description`, `position`) VALUES
-  (NULL, 'displayFooterCategory', 'Category footer', 'This hook adds new blocks under the products listing in a category/search', '1'),
```
Comment (jolelievre) [inline install-dev/upgrade/sql/1.7.8.0.sql:5]: Nice!

**Review by jolelievre — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by PierreRambaud** [comment]
Thanks @matks

**Pull request merged**